polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala

```diff
@@ -379,59 +398,65 @@ object SocketTransport {
     }
 
     def read(): TaskB[Option[Option[ByteBuffer]]] = effectBlocking(readBuffer()).uninterruptible.catchSome {
-      case err: AsynchronousCloseException => Logging.info("Remote peer closed connection") *> ZIO.succeed(None)
-    }.onError {
-      err => Logging.error(s"Connection error ${err.getClass.getName}", err)
+      case err: AsynchronousCloseException => Logging.info("Remote peer closed connection") *> close() *> ZIO.succeed(None)
+    }.catchSome {
+      case err if err.getClass.getSimpleName == "AsynchronousCloseException" => Logging.info("WTFBRO") *> close() *> ZIO.succeed(None)
```

Comment:
🤣

Reply:
Ha ha, good catch... was struggling to figure out why the `onError` was still happening even when it was an `AsyncCloseException` that should have been caught. Changing to `tapError` fixed it, but forgot to remove this bit. Will clean up 😳